runtime/frequency/src/lib.rs

```diff
@@ -421,6 +491,106 @@ impl<T: pallet_collator_selection::Config> OnRuntimeUpgrade for MigratePalletsCu
 	}
 }
 
+/// Migration to set the initial safe XCM version for the XCM pallet.
+#[cfg(feature = "frequency-bridging")]
+pub struct SetSafeXcmVersion<T>(core::marker::PhantomData<T>);
+
+#[cfg(feature = "frequency-bridging")]
+use common_runtime::constants::xcm_version::SAFE_XCM_VERSION;
+
+#[cfg(feature = "frequency-bridging")]
+impl<T: pallet_xcm::Config> OnRuntimeUpgrade for SetSafeXcmVersion<T> {
+	fn on_runtime_upgrade() -> Weight {
+		use sp_core::Get;
+
+		// Access storage directly using storage key because `pallet_xcm` does not provide a direct API to get the safe XCM version.
+		let storage_key = frame_support::storage::storage_prefix(b"PolkadotXcm", b"SafeXcmVersion");
+		log::info!("Checking SafeXcmVersion in storage with key: {:?}", storage_key);
+
+		let current_version = frame_support::storage::unhashed::get::<u32>(&storage_key);
+		match current_version {
+			Some(version) if version == SAFE_XCM_VERSION => {
+				log::info!("SafeXcmVersion already set to {}, skipping migration.", version);
+				T::DbWeight::get().reads(1)
+			},
+			Some(version) => {
+				log::info!(
+					"SafeXcmVersion currently set to {}, updating to {}",
+					version,
+					SAFE_XCM_VERSION
+				);
+				// Set the safe XCM version directly in storage
+				frame_support::storage::unhashed::put(&storage_key, &(SAFE_XCM_VERSION));
+				T::DbWeight::get().reads(1).saturating_add(T::DbWeight::get().writes(1))
+			},
```

Comment:
This appears to allow both upgrading and downgrading the safe XCM version. Just confirming we want to allow both?